README.md

```diff
@@ -34,89 +63,165 @@ message MyContext {
+
+### 2. Client side setup:
+
+The client side setup just has one step. Simply add the `KonigKontextClientInterceptor` to all of your gRPC clients and
+pass in your previously defined KonigKontext instance:
+
+Kotlin:
+
 ```kotlin
-val myServiceServer = ServerBuilder
-    .forPort(<your_port_here>)
-    .addKonigKontextServer(MyService(), MyContext::class)
+val myServiceClient = MyServiceCoroutineStub(myManagedChannel)
+    .withInterceptors(KonigKontextClientInterceptor(MY_GLOBAL_KONTEXT))
+
+// Or for a slightly more succinct way you can optionally use the Kotlin idiomatic helper function instead:
+val myServiceClient = MyServiceCoroutineStub(myManagedChannel)
+    .withKonigKontextInterceptor(MY_GLOBAL_KONTEXT)
+```
+
+Java:
+
+```java
+MyServiceBlockingStub myServiceClient=MyServiceGrpc
+        .newBlockingStub(ManagedChannelBuilder.forTarget("port here").usePlaintext().build())
+        .withInterceptors(new KonigKontextClientInterceptor<>(GlobalKontext.MY_GLOBAL_KONTEXT));
+```
+
+### 3. Server side setup:
+
+The server side setup also just has one step. Simply add the `KonigKontextServerInterceptor` to all of your gRPC servers
+and pass in your previously defined KonigKontext instance:
+
+Kotlin:
+
+```kotlin
+val myServer = ServerBuilder
+    .forPort(<port here>)
+    .addService(MyService())
+    .intercept(KonigKontextServerInterceptor(MY_GLOBAL_KONTEXT))
     .build()
+
+// Or for a slightly more succinct way you can optionally use the Kotlin idiomatic helper function instead:
+val myServer = ServerBuilder
+    .forPort(<port here>)
+    .addService(MyService())
+    .withKonigKontextInterceptor(MY_GLOBAL_KONTEXT)
+    .build() 
+```
+
+Java:
+
+```java
+Server myServer = ServerBuilder
+        .forPort(<port here>)
+        .addService(new MyService())
+        .intercept(new KonigKontextServerInterceptor<>(GlobalKontext.MY_GLOBAL_KONTEXT))
+        .build();
 ```
 
 That's it! No more setup required. See the Usage section below for next steps on actually using the library.
 
 Usage
 -----
 
-### Setting Konig Kontext
+### Setting KonigKontext
 
-To update the Konig Kontext, call `withKonigKontext` supplying your updated context and a closure containing the remaining
-code in the request path:
+Setting the KonigKontext differs depending on the language you are using. See the following snippets for your language:
+
+Kotlin:
 ```kotlin
-val myNewContext = myContext { 
-    // Set your fields here
+withKonigKontext(MY_GLOBAL_KONTEXT.setValue(/* set value here */)) {
+    // Remaining code path that will have access to the updated KonigKontext
 }
+```
 
-withKonigKontext(myNewContext) {
-    // Remaining code path that will have access to the KonigKontext
-}
+Java:
+```java
+GlobalKontext.MY_GLOBAL_KONTEXT.setValue(/* set value here */).run(() -> {
+    // Remaining code path that will have access to the updated KonigKontext 
+})
```

Comment:
This is the one part of the API that I don't love at the moment. Open to suggestions here. I tried looking at CoroutineContext and gRPC Context as examples for both Kotlin and Java based context APIs. For CoroutineContext in Kotlin the API looks like:
```kotlin
val newCoroutineContext = coroutineContext.plus(MyCoroutineContextElement(someValues))

withCoroutineContext(newCoroutineContext) {
  // code with access to newCoroutineContext
}
```

And for Context in java it looks like:
```java
Context newContext = Context.current().withValue(myContextKey, someValue);

newContext.run(() -> {
  // code with access to newCoroutineContext. This lambda is void so we can't assign the result to a variable directly
});
```

The solution here seemed like the closest I could get to mimicking those APIs, but I still don't love the ergonomics. Thoughts?